### PR TITLE
DMP-2819 Implement GET event mappings endpoint

### PR DESCRIPTION
**Pull request by jackmaloney**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-2819

### Change description ###
Implement the GET /event-mappings/{id} endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/event/controller/EventsController.java [hunk 1]
```diff
@@ -90,7 +94,17 @@ public ResponseEntity<List<CourtLog>> courtlogsGet(
         List<CourtLog> courtLogs = courtLogsService.getCourtLogs(courthouse, caseNumber, startDateTime, endDateTime);
 
         return new ResponseEntity<>(courtLogs, HttpStatus.OK);
+    }
+
+    @Override
+    @SecurityRequirement(name = SECURITY_SCHEMES_BEARER_AUTH)
+    @Authorisation(contextId = ANY_ENTITY_ID,
+        globalAccessSecurityRoles = {SUPER_ADMIN})
+    public ResponseEntity<EventMapping> eventMappingsEventHandlerIdGet(Integer eventHandlerId) {
```
Comment (tonymort) [inline src/main/java/uk/gov/hmcts/darts/event/controller/EventsController.java:103]: You can give the controller method names a nicer name by specifying the `operationId` in the open api spec

File: src/test/java/uk/gov/hmcts/darts/event/service/impl/EventMappingServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class EventMappingServiceImplTest {
+
+    @Mock
+    EventHandlerRepository eventHandlerRepository;
+
+    @InjectMocks
+    EventMappingServiceImpl eventDispatcher;
```
Comment (tonymort) [inline src/test/java/uk/gov/hmcts/darts/event/service/impl/EventMappingServiceImplTest.java:29]: Field name needs changing

File: tempWorkspace/3ba664d4-3c18-4361-b1b9-d1e31ed8fad7
Comment (tonymort) [inline tempWorkspace/3ba664d4-3c18-4361-b1b9-d1e31ed8fad7:1]: This needed?

**Review by tonymort — COMMENTED** [review]
few minor things

File: src/integrationTest/java/uk/gov/hmcts/darts/event/controller/EventControllerGetEventsMappingTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            .andExpect(jsonPath("$.type", Matchers.is("99999")))
+            .andExpect(jsonPath("$.sub_type", Matchers.is("8888")))
+            .andExpect(jsonPath("$.name", Matchers.is("some-desc")))
+            .andExpect(jsonPath("$.handler", Matchers.is("Dummy integration test handler")))
+            .andExpect(jsonPath("$.is_active", Matchers.is(true)))
+            .andExpect(jsonPath("$.has_restrictions", Matchers.is(false)))
+            .andExpect(jsonPath("$.created_at").exists());
+    }
+
+    @Test
```
Comment (tonymort) [inline src/integrationTest/java/uk/gov/hmcts/darts/event/controller/EventControllerGetEventsMappingTest.java:56]: See https://github.com/hmcts/darts-api/blob/master/src/integrationTest/java/uk/gov/hmcts/darts/task/service/AdminPatchAutomatedTaskTest.java for quick way to test all roles without permissions are forbidden.  Up to you, think its only me who does this.

File: tempWorkspace/3ba664d4-3c18-4361-b1b9-d1e31ed8fad7
Comment (tonymort) [inline tempWorkspace/3ba664d4-3c18-4361-b1b9-d1e31ed8fad7:1]: This needed?

File: src/test/java/uk/gov/hmcts/darts/event/service/impl/EventMappingServiceImplTest.java (hunk 2 above)
Comment (jackmaloney) [reply src/test/java/uk/gov/hmcts/darts/event/service/impl/EventMappingServiceImplTest.java:29]: Good spot

**Review by tonymort — APPROVED** [review]

**Pull request merged**